resolwe_bio/processes/variant_calling/samtools.yml

```diff
@@ -30,38 +45,42 @@
     runtime: polyglot
     language: bash
     program: |
-      echo "uncompressing genome, indexing"
       GENOME_NAME=`basename '{{ genome.fasta.file }}' .fasta.gz`
-      cp {{ genome.fasta.file }} "${GENOME_NAME}.fasta.gz"
-      echo "{\"proc.progress\":0.05,\"proc.rc\":$?}"
-
-      samtools faidx "${GENOME_NAME}.fasta.gz"
-      echo "{\"proc.progress\":0.1,\"proc.rc\":$?}"
-
-      echo "bam files processing"
       BAM_FILE=`basename '{{ mapping.bam.file }}' .bam`
-      echo "{\"proc.progress\":0.3,\"proc.rc\":$?}"
 
-      echo "removing duplicates"
-      samtools rmdup "{{ mapping.bam.file }}" "${BAM_FILE}_nodupli.bam"
-      echo "{\"proc.progress\":0.4,\"proc.rc\":$?}"
+      gzip -d -c "{{ genome.fasta.file }}" > "${GENOME_NAME}.fasta"
+      re-progress 0.05
+
+      samtools faidx "${GENOME_NAME}.fasta"
+      re-checkrc "Indexing of reference sequence file failed."
+      re-progress 0.1
 
-      echo "creating VCF file"
-      samtools mpileup -u -f "${GENOME_NAME}.fasta.gz" "${BAM_FILE}_nodupli.bam" > "${BAM_FILE}_raw.bcf";
-      echo "{\"proc.progress\":0.5,\"proc.rc\":$?}"
+      {% if options.rmdup %}
+        echo "removing duplicates"
+        samtools rmdup "{{ mapping.bam.file }}" "${BAM_FILE}_dedup.bam"
+        re-checkrc "The process of removing duplicates failed."
+        samtools mpileup -Bugf "${GENOME_NAME}.fasta" "${BAM_FILE}_dedup.bam" {% if bed %} -l "{{ bed.BED.file }}" {% endif %} > "${BAM_FILE}_raw.bcf"
```

Comment:
Following three lines are almost the same as those in else. Put them out of if and in if just define which bam file to use.